lib/mix/tasks/phx.gen.live.ex

```diff
@@ -0,0 +1,220 @@
+      {:new_eex, "live_helpers.ex", Path.join([web_prefix, "live", "live_helpers.ex"])},
+    ]
+  end
+
+  defp copy_new_files(%Context{} = context, binding, paths) do
+    files = files_to_be_generated(context)
+    Mix.Phoenix.copy_from(paths, "priv/templates/phx.gen.live", binding, files)
+    if context.generate?, do: Gen.Context.copy_new_files(context, paths, binding)
+
+    context
+  end
+
+  defp maybe_inject_helpers(%Context{context_app: ctx_app} = context) do
+    web_prefix = Mix.Phoenix.web_path(ctx_app)
+    [lib_prefix, web_dir] = Path.split(web_prefix)
+    file_path = Path.join(lib_prefix, "#{web_dir}.ex")
+    file = File.read!(file_path)
+    inject = "import #{inspect(context.web_module)}.LiveHelpers"
+
+    if String.contains?(file, inject) do
+      :ok
+    else
+      do_inject_helpers(context, file, file_path, inject)
+    end
+
+    context
+  end
+
+  defp do_inject_helpers(context, file, file_path, inject) do
+    Mix.shell().info([:green, "* injecting ", :reset, Path.relative_to_cwd(file_path)])
+
+    new_file = String.replace(file, "import Phoenix.LiveView.Helpers", "import Phoenix.LiveView.Helpers\n      #{inject}")
+    if file != new_file do
+      File.write!(file_path, new_file)
+    else
+      Mix.shell().info """
+
+      Add your #{inspect(context.web_module)}.LiveHelpers import to your view_helpers block in #{file_path}:
+
+          defp view_helpers do
+            quote do
+              # Use all HTML functionality (forms, tags, etc)
+              use Phoenix.HTML
+
+              # Import convenience functions for LiveView rendering
+              import Phoenix.LiveView.Helpers
+              #{inject}
+              ...
+            end
+          end
+      """
+    end
+  end
+
+  @doc false
+  def print_shell_instructions(%Context{schema: schema, context_app: ctx_app} = context) do
+    if schema.web_namespace do
+      Mix.shell().info """
+
+      Add the live routes to your #{schema.web_namespace} :browser scope in #{Mix.Phoenix.web_path(ctx_app)}/router.ex:
+
+          scope "/#{schema.web_path}", #{inspect Module.concat(context.web_module, schema.web_namespace)}, as: :#{schema.web_path} do
+            pipe_through :browser
+            ...
+
+            live "/#{schema.plural}", #{inspect(schema.alias)}Live.Index, :index
+            live "/#{schema.plural}/new", #{inspect(schema.alias)}Live.Index, :new
+            live "/#{schema.plural}/:id/edit", #{inspect(schema.alias)}Live.Index, :edit
+
+            live "/#{schema.plural}/:id", #{inspect(schema.alias)}Live.Show, :show
+            live "/#{schema.plural}/:id/show/edit", #{inspect(schema.alias)}Live.Show, :edit
+          end
+      """
+    else
+      Mix.shell().info """
+
+      Add the live routes to your browser scope in #{Mix.Phoenix.web_path(ctx_app)}/router.ex:
+
+          live "/#{schema.plural}", #{inspect(schema.alias)}Live.Index, :index
```

Comment:
The routes part is the same on both `if/else` branches, should we should probably avoid this duplication. If you want to avoid extra new lines, remember you can always use `\` at the last element in a new line in a string/heredoc to skip a newline.